Assets/Scripts/Environment/IslandHeart/IslandHeartLeveling.cs

```diff
@@ -0,0 +1,156 @@
+using Codice.CM.Common.Tree.Partial;
+using Player;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Transactions;
+using UnityEngine;
+using UnityEngine.Android;
+
+public class IslandHeartLeveling : MonoBehaviour
+{
+    /// <summary>
+    /// This event must be subscribed to by all scripts that need to trigger when the
+    /// Island Heart this script is attached to levels up
+    /// </summary>
+    public event EventHandler islandHeartLevelUp;
+
+	[Tooltip("The world space where the generator is placed when attached to Island Heart")]
+	[SerializeField]
+	private GameObject _generatorLocation;
+	private GameObject _generator;
+
+	[Header("Island Heart Level Manager")]
+	[Tooltip("Current Island Heart Level")]
+	[Min(1f)]
+	[SerializeField]
+	private float _heartLevel = 1;
+
+    [Tooltip("Maximum Island Heart Level")]
+    [Min(2f)]
+    [SerializeField]
+    private float _maxHeartLevel = 2;
+
+	[Tooltip("Multiplier to increase the experience threshold to level up the Island Heart")]
+	[Min(1)]
+	[SerializeField]
+	private float _xpMultiplier = 2;
+
+	[Tooltip("The experience threshold to level up the Island Heart")]
+	[field: SerializeField]
+	public float _xpThreshold
+	{
+		get;
+		private set;
+	} = 5;
+
+	[Tooltip("The current amount of experience the Island Heart has")]
+    [field: SerializeField]
+    public float _currentXP
+    {
+        get;
+        private set;
+    } = 0;
+
+
+    // Start is called before the first frame update
+    void Start()
+	{
+		_generator = null;
+	}
+
+	public void OnTriggerEnter(Collider other)
+	{
+		if(other.gameObject.GetComponent<GeneratorHandler>() && _generator == null)
+		{
+			GeneratorHandler handler = other.gameObject.GetComponent<GeneratorHandler>();
+			_generator = handler.gameObject;
+			handler.SetCharge((_currentXP/_xpThreshold) *100);
+			_generator.transform.parent = _generatorLocation.transform;
+			_generator.GetComponent<Rigidbody>().isKinematic = true;
+		} else if(other.gameObject.tag.Equals("JellyDew") && !IsMaxLevel())
```

Comment:
I tried dropping jelly dew on / near the island heart and it didn't do anything.